File: Python/mergetwo.py
# Definition for singly-linked list.
class ListNode(object):
    def __init__(self, val=0, next=None):
         self.val = val
         self.next = next
    
    def append(self, val):
        current = self
        while current.next is not None:
            current = current.next
        current.next = ListNode(val)

    def print_linked_list(node):
        while node is not None:
            print(node.val, end=" ")
            node = node.next
        print()

class Solution:
    def merging(self, a: ListNode, b: ListNode) -> ListNode:
        head = ListNode(0)
        handler = head
        
        while a is not None and b is not None:
            if a.val < b.val:
                handler.next = a
                a = a.next
            else:
                handler.next = b
                b = b.next
            handler = handler.next

        if a is not None:
            handler.next = a
        elif b is not None:
            handler.next = b

        return head.next
    






solution = Solution()

input_one = [1,4,7]
a = ListNode(input_one[0])  
for i in input_one[1:]:
    a.append(i)

input_two = [2,3,5]
b = ListNode(input_two[0])
for j in input_two[1:]:
    b.append(j)

result = solution.merging(a, b)


ListNode.print_linked_list(result)

File: Java/mergetwo.java
public class mergetwo {
    // Define the Node class
    static class Node {
        int val;
        Node next;

        Node(int val) {
            this.val = val;
        }
    }

    public Node merging(Node a, Node b) {
        Node head = new Node(0);
        Node handler = head;

        while(a != null && b != null) {
            if(a.val < b.val) {
                handler.next = a;
                a = a.next;
            } else {
                handler.next = b;
                b = b.next;
            }
            handler = handler.next;
        }

        if(a != null) {
            handler.next = a;
        } else if(b != null) {
            handler.next = b;
        }

        return head.next;
    }

    public static void main(String[] args) {
        mergetwo morg = new mergetwo();

        // Create the first list
        Node l1 = new Node(1);
        l1.next = new Node(4);
        l1.next.next = new Node(7);

        // Create the second list
        Node l2 = new Node(2);
        l2.next = new Node(3);
        l2.next.next = new Node(5);
        l2.next.next.next = new Node(6);

        // Merge the lists
        Node result = morg.merging(l1, l2);

        // Print the result
        while (result != null) {
            System.out.print(result.val + " ");
            result = result.next;
        }
    }
}